cli/azd/cmd/infra_create.go

```diff
@@ -72,7 +72,7 @@ func (ica *infraCreateAction) Run(ctx context.Context, cmd *cobra.Command, args
 		return fmt.Errorf("loading environment: %w", err)
 	}
 
-	_, err = project.LoadProjectConfig(azdCtx.ProjectPath(), &environment.Environment{})
+	proj, err := project.LoadProjectConfig(azdCtx.ProjectPath(), &environment.Environment{})
```

Comment:
On potential idea is to introduce an `Initialize()` or similar method on `ProjectConfig` that gets called before we return the instance back from `project.LoadProjectConfig(...)`.  

This could then internally loop through all the project services and initialize the framework services for each project.  

This is more work, but would align better to keep all language specific functionality together vs having special uses cases scattered throughout the codebase.

Reply:
`ServiceConfig` has a `GetFrameworkService(..)` method that you can call during new `Initialize()` method.  We can then add an `Initialize()` or similar method onto the `FrameworkService` interface.

Then in the .NET framework service it can call the secret init.